bin/src/Main.purs

```diff
@@ -529,22 +531,34 @@ main = do
               }
             void $ runSpago env (Sources.run { json: args.json })
           Init args@{ useSolver } -> do
-            -- Figure out the package name from the current dir
-            let candidateName = fromMaybe (String.take 50 $ Path.basename Paths.cwd) args.name
-            logDebug [ show Paths.cwd, show candidateName ]
-            packageName <- case PackageName.parse (PackageName.stripPureScriptPrefix candidateName) of
-              Left err -> die [ "Could not figure out a name for the new package. Error:", show err ]
-              Right p -> pure p
+            when (isJust args.name && isJust args.subpackage) $ die [ "--name and --subpackage may not be given at the same time." ]
```

Comment:
Mm... I'd expect the subpackage and folder in which it's created to often be the same thing. So, writing `spago init --name bar/bar` would be annoying to create the `bar` subpackage within the `bar` directory.

Reply:
Ah yes, I implied that there would be a `spago.yaml` created inside the `libs/bar` folder - but the main idea still applies, we'd use slashes to detect if the user wants a root package or a subpackage

Reply:
I see the issue with the above now - creating a root package means populating the current folder, while subpackage creates folders, and that adds ambiguity.

I guess we can live with `spago init --name bar --directory bar`?
Note that it's equivalent to `spago init --name bar/bar` - we wouldn't want to try to infer directory names (i.e. automatically adding the `bar` folder for a package called `bar`), because that very much depends on what structure the user would like, and we shouldn't impose any judgements on that part.

Reply:
Since other file system stuff will need to be modified by the user manually anyway, why not just use a ‘--subpackage’ flag rather than a slash within the name arg to indicate we're creating a subpackage?

Reply:
My above comment proposes a `--directory` flag instead of a `subpackage` flag. The latter makes it clear that we are creating a subpackage, but the former is clear about where the subpackage is going to be, which I find more useful

Reply:
You gave an example `spago init --name bar --directory bar`. What would be the meaning when the two parameters don't have the same value? What is supposed to happen then? For example `spago init --name foo --directory bar`

Reply:
That would create a subpackage called `foo` inside the directory `bar`.

Reply:
What should happen when the `--name` parameter is not given but the `--directory` parameter is?

Reply:
I'd fail then.